model/DataQuery.php

```diff
@@ -661,6 +664,15 @@ public function applyRelation($relation) {
 					$this->query->addLeftJoin($componentBaseClass,
 						"\"$relationTable\".\"$componentField\" = \"$componentBaseClass\".\"ID\"");
 				}
+				else {
+					$alias = uniqid($componentBaseClass);
+					$this->query->addLeftJoin(
+						$componentBaseClass,
+						"\"$relationTable\".\"$componentField\" = \"$alias\".\"ID\"",
+						$alias
+					);
+					$this->lastAlias = $alias;
+				}
```

Comment:
The `isJoinedTo` is going to need to be replaced with something that checks not only the class, but also the alias of the class is correct. Otherwise, cyclic joins aren't going to work. `->applyRelation('Page.Parent.Parent`)`